### PR TITLE
feat: expose delete commands to scratch orgs

### PR DESCRIPTION
**Pull request by RitamAgrawal**
### What does this PR do?
Exposes `SFDX: Delete This from Project and Org` & `SFDX: Delete from Project and Org` 
commands to scratch orgs. Fixes source tracking information for scratch orgs as well as source tracked sandboxes.
### What issues does this PR fix or reference?
<Insert GitHub Issue>@W-12757241@

### Functionality Before
- Commands `SFDX: Delete This from Project and Org` & `SFDX: Delete from Project and Org` were not exposed to scratch orgs
- These commands ran and did not update source tracking information in the source-tracked sandbox.
### Functionality After
- Commands `SFDX: Delete This from Project and Org` & `SFDX: Delete from Project and Org` are now exposed in Command palette as well as context menu in scratch orgs.
- Commands run with an additional argument `--tracksource` to update the source tracking information.

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -218,7 +218,7 @@
         },
         {
           "command": "sfdx.force.source.delete.current.file",
-          "when": "sfdx:project_opened && sfdx:default_username_has_no_change_tracking && resourceLangId != 'forcesourcemanifest'"
+          "when": "sfdx:project_opened && resourceLangId != 'forcesourcemanifest' && editorIsOpen"
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-core/package.json:221]: Why add " && editorIsOpen" here?  Did removing only "sfdx:default_username_has_no_change_tracking && " not work correctly?

Comment (RitamAgrawal) [reply packages/salesforcedx-vscode-core/package.json:221]: This command is provided for the file opened in the editor. That's why added `&& editorIsOpen`

Comment (RitamAgrawal) [reply packages/salesforcedx-vscode-core/package.json:221]: This object is listd under editor/context, so `&& editorIsOpen` should be removed.

File: packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceDelete.test.ts [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe('Force Source Delete', () => {
     const sourcePath = path.join('example', 'path');
     const sourceDeleteCommand = executor.build({ filePath: sourcePath });
     expect(sourceDeleteCommand.toCommand()).to.equal(
-      `sfdx force:source:delete --sourcepath ${sourcePath} --noprompt`
+      `sfdx force:source:delete --sourcepath ${sourcePath} --noprompt --tracksource`
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceDelete.test.ts:30]: 👍🏽

Comment (klewis-sfdc) [reply packages/salesforcedx-vscode-core/test/vscode-integration/commands/forceSourceDelete.test.ts:30]: The changes in the PR are looking great!  What do you think about writing a new jest test to replace this test?

**Comment by klewis-sfdc** [comment]
This is looking great, @RitamAgrawal !  I started doing a little QA and saw that the CLI complains if you use the `--tracksource` flag when not connected to a source-tracked org.  




You should be able to call `getWorkspaceOrgType` first and only add the `--tracksource` flag if the OrgType is SourceTracked.  

Let me know if you have any questions or want to take a look together.  Thanks for this work so far!

File: packages/salesforcedx-vscode-core/test/jest/commands/delete/forceSourceDelete.test.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import { ForceSourceDeleteExecutor } from '../../../../src/commands';
+
+describe('Force Source Delete', () => {
+  let isSourceTracked: boolean;
+  const sourcePath: string = path.join('example', 'path');
+  it('Should build the source delete command with --tracksource flag', () => {
```
Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-core/test/jest/commands/delete/forceSourceDelete.test.ts:14]: nit: can you put an empty line before each test to make the file a bit easier to read?

Comment (klewis-sfdc) [inline packages/salesforcedx-vscode-core/test/jest/commands/delete/forceSourceDelete.test.ts:14]:
```suggestion
  it('Should build the source delete command with --tracksource flag when connected to a source-tracked org', () => {
```

**Comment by klewis-sfdc** [comment]
Hi @RitamAgrawal,

This PR looks good!  I have a few suggested edits to the test file, and the branch needs to be updated with latest from develop, but other than that, this looks good and is ready to go once these last small changes are in!

QA looks good:
✅  Delete operations fix source-tracking appropriately when the SFDX: Delete* commands are used against a source-tracked Org (source-tracked Sandbox and Scratch Org)
✅  SFDX: Delete* commands still behave as before and delete components as expected against a non-source-tracked org

**Review by klewis-sfdc — APPROVED** [review]
Thank you for making those updates!  Looks great.  Thank you.  

After the branch is caught up with the latest from develop, and the integration tests re-run to get past that timeout:

:shipit:

**Pull request merged**